src/crashstats_tools/cmd_supersearchfacet.py

```diff
@@ -289,176 +348,86 @@ def supersearchfacet(
-                click.echo(line)
-
-        elif format_type == "tab":
-            # NOTE(willkg): We need to use click.echo because console.print
-            # does rich fancy-stuff with the output
-            for line in tableize_tab(headers=headers, data=records):
-                click.echo(line)
-
-        elif format_type == "markdown":
-            for line in tableize_markdown(headers=headers, data=records):
-                click.echo(line)
-
-        elif format_type == "json":
-            console.print_json(json.dumps(records))
-
-        return
-
-    # If it is in count/period mode, then we have to do one facet for each
-    # period and compose the results.
-
-    # Figure out what facet we're doing
-    facet_name = params["_facets"][0]
-
-    # Map of facet_name -> (map of date -> (map of value -> count))
-    facet_tables = {facet_name: {}}
-
-    # FIXME(willkg): for "period=weekly", does it make sense to anchor it on
-    # beginning of the week? (sunday or monday)
-
-    for day_start, day_end in generate_periods(period, start_date, end_date):
-        params.update({"date": [">=%s" % day_start, "<%s" % day_end]})
-
-        if verbose:
-            console.print(f"Params: {params}")
+    if verbose:
+        console.print(f"Params: {params}")
+
+    facet_data_payload = fetch_supersearch_facets(
+        console=console,
+        host=host,
+        params=params,
+        api_token=api_token,
+        verbose=verbose,
+    )
+
+    if (
+        "signature" not in params["_facets"]
+        and "signature" in facet_data_payload["facets"]
+    ):
+        # The Super Search API adds a "signature" facet by default if no other
+        # "_facets" are specified even when you don't want it--this removes it
+        del facet_data_payload["facets"]["signature"]
+
+    if format_type == "raw":
+        console.print_json(json.dumps(facet_data_payload))
+        return 0
+
+    total = facet_data_payload["total"]
+    facets = facet_data_payload["facets"]
+
+    # We want to print a blank line between things, so we print a blank line
+    # before any thing that's not the first thing
+    first_thing = True
+
+    for facet_name in facets.keys():
+        if facet_name.startswith("cardinality"):
+            facet_tables = convert_cardinality_data(facets, facet_name, total)
+
+        elif facet_name.startswith("histogram"):
+            facet_tables = convert_histogram_data(facets, facet_name, total)
 
-        facet_data = fetch_supersearch_facets(
-            console=console,
-            host=host,
-            params=params,
-            api_token=api_token,
-            verbose=verbose,
-        )
+        else:
+            facet_tables = convert_facet_data(facets, facet_name, total)
 
-        total = remaining = facet_data["total"]
-        facets = facet_data["facets"]
+        if format_type == "json":
+            console.print_json(json.dumps(facet_tables))
+            continue
```

Comment:
For JSON format, it's better to print the whole thing rather than break it up into fields, so we do that here. Further, we probably don't want the blank line between items.